doc/specs/#5000 - Process Model 2.0/#1032 - Elevation Quality of Life Improvements.md

```diff
@@ -0,0 +1,606 @@
+
+## Potential Issues
+
+<table>
+<tr>
+<td><strong>Accessibility</strong></td>
+<td>
+
+The set of changes proposed here are not expected to introduce any new
+accessibility issues. Users can already create elevated Terminal windows. Making
+it easier to create these windows doesn't really change our accessibility story.
+
+</td>
+</tr>
+<tr>
+<td><strong>Security</strong></td>
+<td>
+
+We won't be doing anything especially unique, so there aren't expected to be any
+substantial security risks associated with these changes. Users can already
+create elevated Terminal windows, so we're not really introducing any new
+functionality, from a security perspective.
+
+We're relying on the inherent security of the `runas` verb of `ShellExecute` to
+prevent any sort of unexpected escalation-of-privilege.
+
+<hr>
+
+One security concern is the fact that the `settings.json` file is currently a
+totally unsecured file. It's completely writable by any medium-IL process. That
+means it's totally possible for a malicious program to change the file. The
+malicious program could find a user's "Elevated PowerShell" profile, and change
+the commandline to `malicious.exe`. The user might then think that their
+"Elevated PowerShell" will run `powershell.exe` elevated, but will actually
+auto-elevate this attacker.
+
+If all we expose to the user is the name of the profile in the UAC dialog, then
+there's no way for the user to be sure that the program that's about to be
+launched is actually what they expect.
+
+To help mitigate this, we should _always_ pass the evaluated `commandline` as a
+part of the call to `ShellExecute`. the arguments that are passed to
+`ShellExecute` are visible to the user, though they need to click the "More
+Details" dropdown to reveal them.
+
+We will need to mitigate this vulnerability regardless of adding support for the
+auto-elevation of individual terminal tabs/panes. If a user is launching the
+Terminal elevated (i.e. from the Win+X menu in Windows 11), then it's possible
+for a malicious program to overwrite the `commandline` of their default profile.
+The user may now unknowingly invoke this malicious program while thinking they
+are simply launching the Terminal.
+
+To deal with this more broadly, we will display a dialog within the Terminal
+window before creating **any** elevated terminal instance. In that dialog, we'll
+display the commandline that will be executed, so the user can very easily
+confirm the commandline.
+
+This will need to happen for all elevated terminal instances. For an elevated
+Windows Terminal window, this means _all_ connections made by the Terminal.
+Every time the user opens a new profile or a new commandline in a pane, we'll
+need to prompt them first to confirm the commandline. This dialog within the
+elevated window will also prevent an attacker from editing the `settings.json`
+file while the user already has an elevated Terminal window open and hijacking a
+profile.

```

Comment:
Could we only do this once per settings.json change?

Particularly, I'm envisioning a process like this:
(0) `settings.json`-->`Custom` Profile: `"commandline": "foo.exe"`
(1) Open WT elevated --> `Custom` profile opens
(2) <we get that dialog you're proposing>
(3) open a new tab with `Custom` profile
(4) <new tab just opens; no dialog>
(5) make _any_ change to `settings.json` (via SUI or directly)
(6) open a new tab with `Custom` profile
(7) <we get that dialog you're proposing>

(Devil's advocate to my proposal) Would this idea fall apart if somebody made a change to `foo.exe`? Or if `foo.exe` was replaced with `pwsh.exe bar.ps1` and `bar.ps1` was updated? Would we have to check all of those for changes?

Reply:
Oh wait... this might be what you say a few lines below...